interface/src/error.rs

```diff
@@ -0,0 +1,242 @@
+                "stake account merge failed due to different authority, lockups or state",
+            ),
+            StakeError::CustodianMissing => f.write_str("custodian address not present"),
+            StakeError::CustodianSignatureMissing => f.write_str("custodian signature not present"),
+            StakeError::InsufficientReferenceVotes => {
+                f.write_str("insufficient voting activity in the reference vote account")
+            }
+            StakeError::VoteAddressMismatch => {
+                f.write_str("stake account is not delegated to the provided vote account")
+            }
+            StakeError::MinimumDelinquentEpochsForDeactivationNotMet => f.write_str(
+                "stake account has not been delinquent for the minimum epochs required for \
+                     deactivation",
+            ),
+            StakeError::InsufficientDelegation => {
+                f.write_str("delegation amount is less than the minimum")
+            }
+            StakeError::RedelegateTransientOrInactiveStake => {
+                f.write_str("stake account with transient or inactive stake cannot be redelegated")
+            }
+            StakeError::RedelegateToSameVoteAccount => {
+                f.write_str("stake redelegation to the same vote account is not permitted")
+            }
+            StakeError::RedelegatedStakeMustFullyActivateBeforeDeactivationIsPermitted => {
+                f.write_str("redelegated stake must be fully activated before deactivation")
+            }
+            StakeError::EpochRewardsActive => f.write_str(
+                "stake action is not permitted while the epoch rewards period is active",
+            ),
+        }
+    }
+}
+
+impl<E> DecodeError<E> for StakeError {
+    fn type_of() -> &'static str {
+        "StakeError"
+    }
+}
+
+impl From<u64> for StakeError {
+    fn from(error: u64) -> Self {
+        match error {
+            0 => StakeError::NoCreditsToRedeem,
+            1 => StakeError::LockupInForce,
+            2 => StakeError::AlreadyDeactivated,
+            3 => StakeError::TooSoonToRedelegate,
+            4 => StakeError::InsufficientStake,
+            5 => StakeError::MergeTransientStake,
+            6 => StakeError::MergeMismatch,
+            7 => StakeError::CustodianMissing,
+            8 => StakeError::CustodianSignatureMissing,
+            9 => StakeError::InsufficientReferenceVotes,
+            10 => StakeError::VoteAddressMismatch,
+            11 => StakeError::MinimumDelinquentEpochsForDeactivationNotMet,
+            12 => StakeError::InsufficientDelegation,
+            13 => StakeError::RedelegateTransientOrInactiveStake,
+            14 => StakeError::RedelegateToSameVoteAccount,
+            15 => StakeError::RedelegatedStakeMustFullyActivateBeforeDeactivationIsPermitted,
+            16 => StakeError::EpochRewardsActive,
+            _ => panic!("unsupported error code"),
+        }
+    }
+}
```

Comment:
Is this implementation needed? I don't see where we had it before. With `FromPrimitive` and `ToPrimitive`, we should be covered, and the `panic!` here doesn't inspire confidence :sweat_smile: